### PR TITLE
Hm33

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: src/main/java/com/movie/store/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    public ShoppingCart addSession(MovieSession movieSession, User user) {
+        log.info("Calling method addSession() in ShoppingCartService");
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
```
Comment (dariiabutorina) [inline src/main/java/com/movie/store/service/impl/ShoppingCartServiceImpl.java]: Doesn't this method have return value type ShoppingCart?

File: src/main/java/com/movie/store/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        return shoppingCartDao.add(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart clear(ShoppingCart shoppingCart) {
+        log.info("Calling method clear() in ShoppingCartService");
+        //  shoppingCart.getTickets().clear();
+        //  Case with deleting tickets from orders_tickets and shopping_-crts_tickets
```
Comment (dariiabutorina) [inline src/main/java/com/movie/store/service/impl/ShoppingCartServiceImpl.java]: Do you need these comments?

File: src/main/java/com/movie/store/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        log.info("Calling method getByUser() in ShoppingCartDao");
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery(
+                    "from ShoppingCart s "
+                            + "left join fetch s.tickets t "
+                            + "join fetch s.user "
```
Comment (dariiabutorina) [inline src/main/java/com/movie/store/dao/impl/ShoppingCartDaoImpl.java:48]: Why fetching user since you've got the `@ManyToOne` connection here?

**Review by dariiabutorina — COMMENTED** [review]
I guess you used to have an abstract DAO class since there are so many DAO implementations classes in this PR
Did you really need to remove it?
Also, left some minor comments

File: src/main/resources/hibernate.cfg.xml [hunk 4]
```diff
@@ -3,18 +3,21 @@
         "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
 <hibernate-configuration>
     <session-factory>
-        <property name="hibernate.generate_statistics">true</property>
+<!--        <property name="hibernate.generate_statistics">true</property>-->
```
Comment (KseniiaMakarova) [inline src/main/resources/hibernate.cfg.xml]: let's remove or uncomment this

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
try making a commit into this branch to refresh the state of this branch comparing to `master`
currently I can see classes from the previous HW here

please pay attention to the mistakes I addressed in our conversation

File: src/main/java/com/movie/store/Main.java [hunk 5]
```diff
@@ -35,14 +37,20 @@ public static void main(String[] args) throws AuthenticationException {
...
         movieSession.setShowTime(dateTime);
         movieSession.setCinemaHall(cinemaHall);
+        MovieSession movieSession1 = new MovieSession();
+        movieSession1.setMovie(movie);
+        LocalDateTime dateTime1 = LocalDateTime.of(2020, 12, 12, 15, 30);
+        movieSession1.setShowTime(dateTime1);
+        movieSession1.setCinemaHall(cinemaHall);
         cinemaHall.setCapacity(100);
-        cinemaHall.setMovieSessions(List.of(movieSession));
+        cinemaHall.setMovieSessions(List.of(movieSession, movieSession1));
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/Main.java]: there should be one-way relationship here
see the UML of this project

File: src/main/java/com/movie/store/Main.java [hunk 6]
```diff
@@ -35,14 +37,20 @@ public static void main(String[] args) throws AuthenticationException {
...
         CinemaHallService cinemaHallService
                 = (CinemaHallService) injector.getInstance(CinemaHallService.class);
         cinemaHallService.add(cinemaHall);
         MovieSessionService movieSessionService =
                 (MovieSessionService) injector.getInstance(MovieSessionService.class);
         movieSessionService.add(movieSession);
+        movieSessionService.add(movieSession1);
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/Main.java:51]: in our conversation I have noticed that `movieSessionService.findAvailableSessions()` should be refactored
please do it

File: src/main/java/com/movie/store/Main.java [hunk 7]
```diff
@@ -64,8 +72,14 @@ public static void main(String[] args) throws AuthenticationException {
         ShoppingCartService shoppingCartService
                 = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
         shoppingCartService.addSession(movieSession, user);
+        shoppingCartService.addSession(movieSession1, user);
         log.info("test getByUser() " + shoppingCartService.getByUser(user));
         ShoppingCart shoppingCart = shoppingCartService.getByUser(user);
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/Main.java]: let's swap these two lines and use local variable for logging

File: src/main/java/com/movie/store/model/Order.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Data
+@Entity
+@EqualsAndHashCode
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/model/Order.java]: is `@EqualsAndHashCode` part of `@Data`?

File: src/main/java/com/movie/store/model/ShoppingCart.java [hunk 9]
```diff
@@ -10,22 +13,25 @@
 import javax.persistence.OneToOne;
 import javax.persistence.Table;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
 @Data
 @Entity
+@EqualsAndHashCode
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/model/ShoppingCart.java]: here as well

File: src/main/java/com/movie/store/model/ShoppingCart.java [hunk 10]
```diff
@@ -10,22 +13,25 @@
...
 @Table(name = "shopping_carts")
 public class ShoppingCart {
     @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/model/ShoppingCart.java]: we don't need this - we have `@MapsId`

File: src/main/java/com/movie/store/model/Ticket.java [hunk 11]
```diff
@@ -8,9 +8,11 @@
 import javax.persistence.ManyToOne;
 import javax.persistence.Table;
 import lombok.Data;
+import lombok.EqualsAndHashCode;
 
 @Data
 @Entity
+@EqualsAndHashCode
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/model/Ticket.java]: is `@EqualsAndHashCode` part of `@Data`?

File: src/main/java/com/movie/store/model/ShoppingCart.java [hunk 12]
```diff
@@ -10,22 +13,25 @@
...
     @OneToMany
     @JoinTable(name = "shopping_carts_tickets",
             joinColumns =
                     { @JoinColumn(name = "shopping_cart_id",
-                            referencedColumnName = "shopping_Cart_id") },
+                            referencedColumnName = "shopping_cart_id") },
             inverseJoinColumns =
                     { @JoinColumn(name = "ticket_id", referencedColumnName = "id") })
-    private List<Ticket> tickets;
+    private Set<Ticket> tickets = new HashSet<>();
```
Comment (KseniiaMakarova) [inline src/main/java/com/movie/store/model/ShoppingCart.java:29]: does it work without replacing `List` with `Set`?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — COMMENTED** [review]
let's also remove `log4j.logger.org.hibernate.SQL=DEBUG` property

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**